odoo/addons/base/models/res_users.py

```diff
@@ -799,8 +803,35 @@ def _compute_session_token(self, sid):
             self.env.registry.clear_cache()
             return False
         data_fields = self.env.cr.fetchone()
+        # create tuple with column name and value, allowing for overrides to manipulate the values
+        cr_description = self.env.cr.description
+        return tuple((column.name, data_fields[index]) for index, column in enumerate(cr_description))
+
+    def _session_token_hash_compute(self, sid, field_values):
+        # Generate hmac key using the column name and its value, only if the value is not None
+        # To avoid invalidating sessions when installing a new feature modifying the session token computation
+        # while not still being used.
+        key_tuple = (tuple(
+            (k, v)
+            for k, v in field_values
+            if (
+                any(u is not None for u in v)
+                if isinstance(v, (list, tuple))  # auth_passkey_key_ids may return a list
+                else v is not None
+            )
+        ),)
```

Comment:
I would change the query for auth_pass_key to return null/none instead of that `[None]` to avoid the overhead here. Even if it means database with passkey installed will see a change in their session token hash meaning they will be disconnected on upgrade. It's a minority and I prefer to have the cleanest code than to handle this case.

Reply:
I can update the SQL query to avoid returning a list containing a null value easily. Unfortunately, doing so will break the legacy computing of session token.

I mean that the older session token computing methods await for a `[None]` when no passkey is set. Modifying the query will return a `None`, which is better for the new code.
But it will break the comparison with older session token and to make sure it works, I'll have to update the `_legacy_session_token_hash_compute` so it cast again the `auth_passkey_ids` to a list containing a None.


So a choice have to be made here :/